src/DocumentTemplate/tests/test_DT_Try.py

```diff
@@ -0,0 +1,221 @@
+            '<dtml-try>'
+            'Variable "name": '
+            '<dtml-var expr="str(100)"> '
+            '<dtml-finally>'
+            'Finally variable: '
+            '<dtml-var expr="str(111)"> '
+            '<dtml-call expr="dummy_list.pop(0)">'
+            '</dtml-try>')
+        dummy_list = [100]
+
+        res = html(dummy_list=dummy_list)
+        expected = ('Variable "name": 100 '
+                    'Finally variable: 111 ')
+        self.assertEqual(res, expected)
+        # 100 got removed in the finally block.
+        self.assertEqual(dummy_list, [110])
+
+    def test_DT_Try__Try__05(self):
+        """It executes the finally block if an exception occurs."""
+        html = self.doc_class(
+            '<dtml-call expr="dummy_list.append(110)">'
+            '<dtml-try>'
+            'Variable "name": '
+            '<dtml-var expr="not_defined"> '
+            '<dtml-finally>'
+            'Finally variable: '
+            '<dtml-call expr="dummy_list.pop(0)"> '
+            '</dtml-try>')
+        dummy_list = [100]
+
+        self.assertRaises(NameError, html, dummy_list=dummy_list)
+        # 100 got removed in the finally block.
+        self.assertEqual(dummy_list, [110])
+
+    def test_DT_Try__Try__06(self):
+        """It allows different exception handlers."""
+        html = self.doc_class(
+            '<dtml-try>'
+            'Variable "name": '
+            '<dtml-var expr="not_defined">'
+            '<dtml-except KeyError>'
+            'Exception variable: '
+            '<dtml-var expr="str(100)">'
+            '<dtml-except NameError>'
+            'Exception variable: '
+            '<dtml-var expr="str(110)">'
+            '</dtml-try>')
+        res = html()
+        expected = ('Exception variable: 110')
+        self.assertEqual(res, expected)
+
+    def test_DT_Try__Try__07(self):
+        """It catches errors with handler for base exception."""
+        html = self.doc_class(
+            '<dtml-try>'
+            'Variable "name": '
+            '<dtml-raise KeyError></dtml-raise>'
+            '<dtml-except NameError>'
+            'Exception variable: '
+            '<dtml-var expr="str(100)">'
+            '<dtml-except LookupError>'
+            'Exception variable: '
+            '<dtml-var expr="str(110)">'
+            '</dtml-try>')
+        res = html()
+        expected = ('Exception variable: 110')
+        self.assertEqual(res, expected)
+
+    def test_DT_Try__Try__08(self):
+        """It raises if no exception handler matches."""
+        html = self.doc_class(
+            '<dtml-try>'
+            'Variable "name": '
+            '<dtml-var expr="not_defined">'
+            '<dtml-except KeyError>'
+            'Exception variable: '
+            '<dtml-var expr="str(100)">'
+            '<dtml-except AttributeError>'
+            'Exception variable: '
+            '<dtml-var expr="str(110)">'
+            '</dtml-try>')
+        self.assertRaises(NameError, html)
```

Comment:
I'd prefer: (here and on the other places where `self.assertRaises` is used)

```python
self.assertRaises(NameError):
    html()
```

Reply:
I refactored it to 
```python
with self.assertRaises(NameError):
    html()
```